lib/active_merchant/billing/gateways/creditcall.rb

```diff
@@ -0,0 +1,204 @@
+        end
+
+        commit(request)
+      end
+
+      def capture(money, authorization, options={})
+        request = build_xml_request do |xml|
+          add_transaction_details(xml, money, authorization, "Conf", options)
+          add_terminal_details(xml, options)
+        end
+
+        commit(request)
+      end
+
+      def refund(money, authorization, options={})
+        request = build_xml_request do |xml|
+          add_transaction_details(xml, money, authorization, "Refund", options)
+          add_terminal_details(xml, options)
+        end
+
+        commit(request)
+      end
+
+      def void(authorization, options={})
+        request = build_xml_request do |xml|
+          add_transaction_details(xml, nil, authorization, "Void", options)
+          add_terminal_details(xml, options)
+        end
+
+        commit(request)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<TransactionKey>).+?(</TransactionKey>))i, '\1[FILTERED]\2').
+          gsub(%r((<PAN>).+?(</PAN>))i, '\1[FILTERED]\2').
+          gsub(%r((<CSC>).+?(</CSC>))i, '\1[FILTERED]\2')
+      end
+
+      private
+
+      def build_xml_request
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.Request(type: "CardEaseXML", version: "1.0.0") do
+            yield(xml)
+          end
+        end
+        builder.to_xml
+      end
+
+      def add_transaction_details(xml, amount, authorization, type, options={})
+        xml.TransactionDetails do
+          xml.MessageType type
+          xml.Amount(unit: "Minor"){ xml.text(amount) } if amount
+          xml.CardEaseReference authorization if authorization
+          xml.VoidReason "01" if type == "Void"
+        end
+      end
+
+      def add_terminal_details(xml, options={})
+        xml.TerminalDetails do
+          xml.TerminalID @options[:terminal_id]
+          xml.TransactionKey @options[:transaction_key]
+          xml.Software(version: "SoftwareVersion"){ xml.text("SoftwareName") }
+        end
+      end
+
+      def add_card_details(xml, payment_method, options={})
+        xml.CardDetails do
+          xml.Manual(type: "cnp") do
+            xml.PAN payment_method.number
+            xml.ExpiryDate(format: "yyMM"){ xml.text(expiration_date(payment_method)) }
+            xml.CSC payment_method.verification_value
+          end
+
+          if address = options[:billing_address] || options[:address]
+            xml.AdditionalVerification do
+              xml.Address address[:address1]
+              xml.Zip address[:zip]
+            end
+          end
+        end
+      end
+
+      def expiration_date(payment_method)
```

Comment:
Can we take advantage of https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/credit_card_formatting.rb#L13 here?